### ISSUE TITLE
The whiteboard cursor doesn't always show the user's name

### ISSUE THREAD

**Issue by capilkey** [bigbluebutton/bigbluebutton#6613]
Sometimes when turning multi-user whiteboard on in the HTML5 client the presenter's cursor doesn't render the presenter's name correctly to everyone. The specific details on reproduction are unclear to me, but I have seen this problem multiple times. In my most recent case I turned multi-user whiteboard on and the cursor rendered correctly, but after toggling it again it broke for just the viewer. 

What the viewer saw:
![image](https://user-images.githubusercontent.com/1395090/51559070-f8aed300-1e4e-11e9-85bf-88d490351781.png)

The presenter displayed the full name as expected and there were no exceptions in the viewer's browser console.

The attached image shows a small empty box where the name should be, but I've also seen it where the box isn't there either. It's unclear to me if it's the same problem or if maybe there are multiple issues with the rendering.

Another problem that might or might not be related is that when the user moves their mouse out of the presentation area the cursor disappears for them, but viewers will still show the last known mouse position so cursors are never removed as they should be.

**Comment by capilkey** [comment]
I found the reproduction steps for the empty name box. They're somewhat specific and require two clients to be open. The steps are:
1. Have multi-user turned off.
2. Move the presenter's mouse slowly off the bottom of the presentation so that the cursor is unrendered from the viewer's screen. (If dragged to the side or too quickly the invalid cursor value won't be sent to the other clients and the cursor won't be deleted)
3. Resize the presenter's screen such that the whiteboard toolbar is outside of the presentation area. See image 1.
4. Move the presenter's mouse into the presentation area so the cursor is rendered.
5. Move the presenter's mouse quickly to the side and out of the presentation area. Make sure the red dot is still visible on the viewer's whiteboard.
6. Without touching the presentation area, toggle multi-user whiteboard on.
7. The empty box should now be visible on the viewer's screen.

If the presenter's cursor is ever unrendered by moving the cursor slowly out the bottom and then back in to the presentation area the name will show up on the viewer's screen.

Image 1.
![image](https://user-images.githubusercontent.com/1395090/51560026-6fe56680-1e51-11e9-9512-1d3fe92f320d.png)

**Comment by capilkey** [comment]
My guess is that there's two issues at play here. The first is that the cursor doesn't have the presenter's name passed to it if multi-user is turned off. The second issue is that in most cases the client doesn't seem to be sending coordinates that will cause the cursor to be unrendered.

In the Flash client I solved the first problem by always passing the name to the cursor whether multi-user whiteboard was turned on or not. The cursor can then decide whether to display the name and there's no need to update the object with new information if multi-user is turned on. I solved the second problem by sending cursor coordinates of (-1, -1) when the cursor is moved out of the presentation area and then on the viewer side if it gets a negative x or y coordinate it either won't render the cursor or the cursor will be invisible. It's unclear if the client isn't sending the negative values or if the client isn't hiding the cursor when receiving negative values.
